resources/aws_ec2.go

```diff
@@ -663,3 +714,365 @@ func (obj *AwsEc2Res) UnmarshalYAML(unmarshal func(interface{}) error) error {
+			err: errwrap.Wrapf(err, "error subscribing to topic"),
+		}:
+		case <-obj.closeChan:
+		}
+		return
+	}
+	log.Printf("%s: Created Subscription", obj)
+}
+
+// snsAuthorize adds the necessary permission for cloudwatch to publish to SNS
+func (obj *AwsEc2Res) snsAuthorize() {
+	gaInput := &sns.GetTopicAttributesInput{
+		TopicArn: aws.String(obj.topicArn),
+	}
+	attrs, err := obj.snsClient.GetTopicAttributes(gaInput)
+	if err != nil {
+		select {
+		case obj.awsChan <- &chanStruct{
+			err: errwrap.Wrapf(err, "error getting topic attributes"),
+		}:
+		case <-obj.closeChan:
+		}
+		return
+	}
+	// unmarshal the current sns publishing policy
+	type statement struct {
+		Sid       string `json:"Sid"`
+		Effect    string `json:"Effect"`
+		Principal *struct {
+			AWS     string `json:"AWS,omitempty"`
+			Service string `json:"Service,omitempty"`
+		} `json:"Principal,omitempty"`
+		Action    interface{} `json:"Action"`
+		Resource  string      `json:"Resource"`
+		Condition *struct {
+			StringEquals *struct {
+				AWSSourceOwner *string `json:"AWS:SourceOwner,omitempty"`
+			} `json:"StringEquals,omitempty"`
+		} `json:"Condition,omitempty"`
+	}
+	type policy struct {
+		Version   string      `json:"Version"`
+		ID        string      `json:"Id"`
+		Statement []statement `json:"Statement"`
+	}
+	var p policy
+	pol := attrs.Attributes["Policy"]
+	err = json.Unmarshal([]byte(*pol), &p)
+	if err != nil {
+		select {
+		case obj.awsChan <- &chanStruct{
+			err: errwrap.Wrapf(err, "error unmarshaling policy"),
+		}:
+		case <-obj.closeChan:
+		}
+		return
+	}
+	// add the topic's arn to the snsPermission json string
+	grantPermission := fmt.Sprintf(snsPermission, obj.topicArn)
+	var s statement
+	err = json.Unmarshal([]byte(grantPermission), &s)
+	if err != nil {
+		select {
+		case obj.awsChan <- &chanStruct{
+			err: errwrap.Wrapf(err, "error unmarshaling snsPermission"),
+		}:
+		case <-obj.closeChan:
+		}
+		return
+	}
+	statements := []statement{
+		p.Statement[0],
+		s,
+	}
+	// inject policy with cloudwatch sns permissions
+	n := p
+	n.Statement = statements
+	newPolicyBytes, err := json.Marshal(n)
+	if err != nil {
+		select {
+		case obj.awsChan <- &chanStruct{
+			err: errwrap.Wrapf(err, "error marshaling policy"),
+		}:
+		case <-obj.closeChan:
+		}
+		return
+	}
+	newPolicy := string(newPolicyBytes)
+	saInput := &sns.SetTopicAttributesInput{
+		AttributeName:  aws.String("Policy"),
+		AttributeValue: &newPolicy,
+		TopicArn:       aws.String(obj.topicArn),
+	}
+	// update the policy
+	_, err = obj.snsClient.SetTopicAttributes(saInput)
+	if err != nil {
+		select {
+		case obj.awsChan <- &chanStruct{
+			err: errwrap.Wrapf(err, "error setting topic attributes"),
+		}:
+		case <-obj.closeChan:
+		}
+		return
+	}
+	log.Printf("%s: Authorized Target", obj)
+}
+
+// Close cleans up when we're done. This is needed to delete some of the AWS
+// objects created for the SNS endpoint.
+func (obj *AwsEc2Res) Close() error {
+	var errors error
```

Comment:
please pick a different var name as this is a common package import. Maybe `errs` or `errList` ?